src/weapon.py

```diff
@@ -160,10 +158,29 @@ def update(
         self.time_since_shoot += delta_time
 
     def shoot(self, projectile_list: arcade.SpriteList):
-        bullet = arcade.SpriteSolidColor(10, 5, arcade.color.RED)
+        bullet: SpriteForMachineGun = SpriteForMachineGun(self)
```

Comment:
why is a machine gun getting assigned to a variable named bullet?

Reply:
Probably `SpriteForMachineGun` is actually a sprite for a machine gun bullet, nor for the gun itself.  Maybe we tweak the naming to make this clearer?

Reply:
I believe the next step will be making classes for these projectiles and beams, so these will all morph into either `SpriteForProjectile` and `SpriteForBeam`, so the naming convention issue will go away